src/Infrastructure.EntityFramework/Repositories/CollectionRepository.cs

```diff
@@ -233,17 +233,42 @@ public async Task ReplaceAsync(Core.Entities.Collection collection, IEnumerable<
         }
     }
 
-    public async Task UpdateUsersAsync(Guid id, IEnumerable<SelectionReadOnly> users)
+    public async Task UpdateUsersAsync(Guid id, IEnumerable<SelectionReadOnly> requestedUsers)
     {
         using (var scope = ServiceScopeFactory.CreateScope())
         {
             var dbContext = GetDatabaseContext(scope);
-            var procedure = new CollectionUserUpdateUsersQuery(id, users);
-            var updateData = await procedure.Update.BuildInMemory(dbContext);
-            dbContext.UpdateRange(updateData);
-            var insertData = await procedure.Insert.BuildInMemory(dbContext);
-            await dbContext.AddRangeAsync(insertData);
-            dbContext.RemoveRange(await procedure.Delete.Run(dbContext).ToListAsync());
+
+            var existingCollectionUsers = await dbContext.CollectionUsers
+                .Where(cu => cu.CollectionId == id)
+                .ToListAsync();
+
+            foreach (var requestedUser in requestedUsers)
+            {
+                var existingCollectionUser = existingCollectionUsers.FirstOrDefault(cu => cu.OrganizationUserId == requestedUser.Id);
+                if (existingCollectionUser == null)
+                {
+                    // This is a brand new entry
+                    dbContext.CollectionUsers.Add(new CollectionUser
+                    {
+                        CollectionId = id,
+                        OrganizationUserId = requestedUser.Id,
+                        HidePasswords = requestedUser.HidePasswords,
+                        ReadOnly = requestedUser.ReadOnly,
+                    });
+                    break;
+                }
+
+                // It already exists, update it
+                existingCollectionUser.HidePasswords = requestedUser.HidePasswords;
+                existingCollectionUser.ReadOnly = requestedUser.ReadOnly;
+                dbContext.CollectionUsers.Update(existingCollectionUser);
+            }
+
+            // Remove all existing ones that are no longer requested
+            var requestedUserIds = requestedUsers.Select(u => u.Id);
+            dbContext.CollectionUsers.RemoveRange(existingCollectionUsers.Where(cu => !requestedUserIds.Contains(cu.OrganizationUserId)));
+            await dbContext.SaveChangesAsync();
         }
```

Comment:
Looking at the existing SQL Server sproc for this, looks like we do a `EXEC [dbo].[User_BumpAccountRevisionDateByCollectionId] @CollectionId, @OrgId`. Doesn't look like we are doing that here. Were we even doing it in the previous IQueryable solution? Maybe we are missing this for a lot of these update functions.

Reply:
I don't think we were doing it for the previous version but I added it to this one.